spec/mail/messagesubmission.mdown

```diff
@@ -0,0 +1,311 @@
+- `notPermittedToSend` – The user does not have permission to send at all right
+  now for some reason. A *description* `String` property MAY be present on the SetError object to display to the user why they are not permitted.
+- `messageNotFound` - The *messageId* is not a valid id for a message in the
+  account.
+- `invalidMessage` - The message to be sent is invalid in some way. The
+  SetError SHOULD contain a property called *properties* of type `String[]` that lists **all** the properties of the Message that were invalid.
+- `invalidProperties` - The object contains properties which are of the wrong
+  type, or is missing a required property, or is specifying a property that may only be set by the server.
+
+If the creation succeeds, the message will be sent to the recipients given in the envelope *rcptTo* parameter. The server MUST remove any *Bcc* header present on the message during delivery. The server MAY add or remove other headers from the submitted message, or make further alterations in accordance with the server's policy during delivery.
+
+If the referenced message is destroyed at any point after the MessageSubmission object is created, this MUST NOT change the behaviour of the message submission (i.e. it does not cancel a future send).
+
+If a *sendAt* property is specified and the date is in the future, the server SHOULD wait until this time before delivering the message. If the server may be able to cancel the send, it MUST return the *maybeCanUnsend* property as `true`. After the message is sent beyond the boundary of the controlling system and can no longer be recalled, this property SHOULD be set to `false` by the server (which will cause a change in the state string for the type, so clients can fetch this as an update).
+
+### Undo send
+
+In some systems, a message that has been queued to be sent may be unsent if it is before the *sendAt* time, or the recipients are all on the same system. The server indicates that it may be possible to cancel the submission by setting the *undoStatus* property to `pending`.
+
+To "unsend" a message, attempt to update the *undoStatus* of the MessageSubmission object from `pending` to `canceled`. If this succeeds, the message will definitely not be sent to any recipients.
+
+If the MessageSubmission exists but the message cannot be unsent, this MUST be rejected with a `cannotUnsend` error. If the message submission object doesn't exist (because the server only retains them for a certain amount of time, or it's just bogus), this MUST be rejected with a `notFound` error.
+
+### Destroying a MessageSubmission object
+
+Destroying a MessageSubmission object MUST NOT affect the delivies it represents. It purely removes the record of the message submission. The JMAP server MAY reject attempts by a client to destroy a MessageSubmission with a `forbidden` error. The server MAY automatically destroy MessageSubmission objects after a certain time or in response to other triggers.
```

Comment:
@jeffpc: Yes, this is a valid (very basic) implementation. As cjnewman said, clients have to be able to deal with these responses anyway.